src/main/java/com/openlattice/chronicle/services/surveys/SurveysManagerImpl.java

```diff
@@ -0,0 +1,467 @@
+         * the caller would get an "ok" response. Instead send an error response.
+         */
+        throw new IllegalArgumentException( "questionnaire not found" );
+    }
+
+    @Override
+    public Map<UUID, Map<FullQualifiedName, Set<Object>>> getStudyQuestionnaires(
+            UUID organizationId, UUID studyId ) {
+        try {
+            logger.info( "Retrieving questionnaires for study :{}", studyId );
+
+            // If an organization does not have the chronicle_surveys app installed, exit early
+            if ( scheduledTasksManager.entitySetIdsByOrgId.get( CHRONICLE_SURVEYS ).containsKey( organizationId ) ) {
+                logger.warn( "No questionnaires found for study {}. {} is not installed on organization with id {}. ",
+                        studyId,
+                        CHRONICLE_SURVEYS.toString(),
+                        organizationId );
+                return ImmutableMap.of();
+            }
+
+            // check if study is valid
+            UUID studyEntityKeyId = Preconditions
+                    .checkNotNull( commonTasksManager.getStudyEntityKeyId( organizationId, studyId ),
+                            "invalid studyId: " + studyId );
+
+            // load apis
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // entity set ids
+            UUID questionnaireESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE_SURVEYS, SURVEY, QUESTIONNAIRE_ES );
+            UUID studiesESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID partOfESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PART_OF, PART_OF_ES );
+
+            checkNotNullUUIDs( ImmutableSet.of( questionnaireESID, studiesESID, partOfESID ) );
+
+            // filtered search on questionnaires ES to get neighbors of study
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi
+                    .executeFilteredEntityNeighborSearch(
+                            studiesESID,
+                            new EntityNeighborsFilter(
+                                    Set.of( studyEntityKeyId ),
+                                    Optional.of( Set.of( questionnaireESID ) ),
+                                    Optional.of( Set.of( studiesESID ) ),
+                                    Optional.of( Set.of( partOfESID ) )
+                            )
+                    );
+
+            // create a mapping from entity key id -> entity details
+            List<NeighborEntityDetails> studyQuestionnaires = neighbors.getOrDefault( studyEntityKeyId, List.of() );
+            Map<UUID, Map<FullQualifiedName, Set<Object>>> result = studyQuestionnaires
+                    .stream()
+                    .filter( neighbor -> neighbor.getNeighborId().isPresent() && neighbor.getNeighborDetails()
```

Comment:
I would avoid `getNeighborId` since the ekid is present on the neighbor entity, and we should be loading entity key ids in a consistent way.